android-lib/src/main/java/com/kinvey/android/store/DataStore.java

```diff
@@ -709,7 +713,7 @@ public void pull(KinveyPullCallback callback) {
      * @param isAutoPagination true if auto-pagination is used
      * @param callback KinveyPullCallback
      */
-    public void pull(Query query, boolean isAutoPagination, KinveyPullCallback callback) {
+    public void pull(@Nullable Query query, boolean isAutoPagination, @NonNull KinveyPullCallback callback) {
         Preconditions.checkNotNull(client, "client must not be null");
```

Comment:
Why is the `Query` allowed to be `null` here?

Reply:
It's old logic. Inside the method, the SDK checks the query is it's null then the new empty query will be created.